.github/workflows/check-cabal-files.yml

```diff
@@ -0,0 +1,31 @@
+name: Haskell Cabal Check
+
+on:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    env:
+      # current ref from: 27.02.2022
+      SECP256K1_REF: ac83be33d0956faf6b7f61a60ab524ef7d6a473a
+
+      SECP_CACHE_VERSION: "2022-12-30"
+
+      PKG_CONFIG_PATH: /usr/lib/pkgconfig:/usr/local/lib/pkgconfig
+
+      LD_LIBRARY_PATH: /usr/local/lib
```

Comment:
Given that the new workflow file is now so small (e.g. this `env:` section can even be still removed I think), what about adding this as a job/step in the existing build workflow? This way, we will have less duplication of e.g. GHC/Cabal versions.

Alternatively, we could just run it via Nix, do you have thoughts on that (see also the [review message](https://github.com/input-output-hk/ouroboros-network/pull/4261#pullrequestreview-1241825272))?

Reply:
It might be advantageous to be able to control the `cabal` version.  For example `cabal check` behaves differently between `cabal-3.6.2.0` and `cabal-3.8.1.0` despite `cabal` being able to compile the project in either case.

Reply:
The `env` is removed now.